README.md

```diff
@@ -97,10 +99,11 @@ const subscriber = store
 subscriber.unsubscribe()
 ```
 
-> NOTE: It's not necessary to fetch state inside of the Provider - but it still needs to be placed somewhere to init the state.
+> NOTE: It's not necessary to fetch state inside the Provider - but it still needs to be placed somewhere to init the state.
 
 ## State props
-The state hook allows you to pass any arguments into the context. It can be some initial state or you can even return it and pass it through to the components. Any state props change will update the context and trigger components rerendering **when necessary**.
+
+The state hook allows you to pass any arguments into the context. It can be some initial state or you could even return it and pass it through to the components via props. All state prop changes will update the context and trigger component re-rendering **only when necessary**.
```

Comment:
```suggestion
The state hook allows you to pass any arguments into the context. It can be some initial state or you could even return it and pass it through to the components. All state prop changes will update the context and trigger component re-rendering **only when necessary**.
```

components gets the state via hook :D